runtime-util/src/test/java/io/camunda/connector/runtime/util/feel/FeelEngineWrapperExpressionEvaluationTest.java

```diff
@@ -49,6 +50,22 @@ void evaluateToJson_ShouldSucceed_WhenHappyCase() {
         .isEqualTo("{\"processedOutput\":{\"statusCode\":\"200 OK\"}}");
   }
 
+  @Test
+  void evaluateToJson_ShouldSucceed_WhenHappyCaseJavaType() {
+    // given
+    // FEEL expression -> {"processedOutput":response.callStatus}
+    final String resultExpression = "{\"processedOutput\": response.callStatus }";
+    // Response from service -> {"callStatus":{"statusCode":"200 OK"}}
+    final Map variables = Map.of("callStatus", Map.of("statusCode", "200 OK"));
+
+    // when
+    final Map evaluatedResultAsMap = objectUnderTest.evaluate(resultExpression, variables);
+
+    // then
+    final Map expectedResult = Map.of("processedOutput", Map.of("statusCode", "200 OK"));
+    assertEquals(expectedResult, evaluatedResultAsMap);
```

Comment:
Let's stay with one Assertion framework if possible
```suggestion
    assertThat(evaluatedResultAsMap).isEqualTo(expectedResult);
```